gprofiler/utils/fs.py

```diff
@@ -27,29 +26,27 @@ def is_rw_exec_dir(path: str) -> bool:
     """
     Is 'path' rw and exec?
     """
-    # randomize the name - this function runs concurrently on paths of in same mnt namespace.
-    test_script = Path(path) / f"t-{token_hex(10)}.sh"
 
     # try creating & writing
     try:
         os.makedirs(path, 0o755, exist_ok=True)
-        test_script.write_text("#!/bin/sh\nexit 0")
-        test_script.chmod(0o755)
+        test_script = NamedTemporaryFile(dir=path, suffix=".sh")
```

Comment:
I think that to have `NamedTemporaryFile` actually remove the file, you need `__exit__` to be called. So the usage should be:
```
with NamedTemporaryFile(...) as test_script:
    do function logic
```

Reply:
I wrote a simple test to check it and it gets closed everytime, even with "sys.exit(1) between these lines:
``` 
tf = tempfile.NamedTemporaryFile(dir="/home/kitter/Documents/june-gprof/my-gprof-fork/gprofiler", suffix=".tester")

with open(tf.name, "w") as f:
    f.write("my-tester")
print(Path(tf.name).read_text())
time.sleep(5)
``` 
but I changed it accordingly anyway :)